integration_test/validate_metadata_test.go

```diff
@@ -64,9 +64,8 @@ func TestThirdPartyPublicUrls(t *testing.T) {
 	err := walkThirdPartyApps(func(contents []byte) error {
 		integrationMetadata := &metadata.IntegrationMetadata{}
 		err := metadata.UnmarshalAndValidate(contents, integrationMetadata)
-		if integrationMetadata.MinimumSupportedAgentVersion == nil {
-			// If the application support has not been released, it will not
-			// have a public URL to validate
+		if integrationMetadata.PublicUrl == "" {
+			// The public doc isn't available yet. 
```

Comment:
Can we assume that `public_url == ""` always implies that `minimum_version` is nil?

Reply:
We can't.
integrationMetadata.PublicUrl == "" covers below three cases:
- app is newly developed and hasn't got released yet. It's in the master branch. thus no MinimumSupportedAgentVersion and no PublicUrl exist yet
- app is newly released thus MinimumSupportedAgentVersion is being filled up by the current PR but no PublicUrl as our doc writer only triggers 3rd party app doc generator post integration release. The generator triggered by google technical writer will create a PR like this https://github.com/GoogleCloudPlatform/ops-agent/pull/866
- if https://github.com/GoogleCloudPlatform/ops-agent/pull/866 was ignored then we end up having no public_url for the app even though it has been public available

Reply:
In that case should we leave the check for `minimum_version`?

```
if integrationMetadata.MinimumSupportedAgentVersion == nil || integrationMetadata.PublicUrl == "" {
```

Reply:
If the third case happens, it will fall inside the if statement on the second condition anyways..

Reply:
Nevermind I got mixed up, LGTM.